File: 2306-create-binary-tree-from-descriptions/create-binary-tree-from-descriptions.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    TreeNode* createBinaryTree(vector<vector<int>>& descriptions) {
     unordered_set<int> childs;
        unordered_map<int, TreeNode*> nodes;

        for (auto& des : descriptions) {
            int p = des[0];
            int c = des[1];
            int type = des[2];
            TreeNode* parent = NULL;
            TreeNode* child = NULL;

            if (nodes.find(p) != nodes.end()) {
                parent = nodes[p];
            } else {
                parent = new TreeNode(p);
                nodes[p] = parent;
            }

            if (nodes.find(c) != nodes.end()) {
                child = nodes[c];
            } else {
                child = new TreeNode(c);
                nodes[c] = child;
            }

            if (type == 1) {
                parent->left = child;
            } else {
                parent->right = child;
            }
            childs.insert(c);
        }

        TreeNode* root = NULL;

        for (auto& pair : nodes) {
            if (childs.find(pair.first) == childs.end()) {
                root = pair.second;
            }
        }

        return root;
    }
};

File: 2306-create-binary-tree-from-descriptions/create-binary-tree-from-descriptions.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public TreeNode createBinaryTree(int[][] descriptions) {
        HashSet<Integer> childs=new HashSet<>();
        HashMap<Integer,TreeNode> nodes=new HashMap<>();

        for(int[] des:descriptions){
          int p=des[0];
          int c=des[1];
          int type=des[2];
          TreeNode parent=null;
          TreeNode child=null;

          if(nodes.containsKey(p)){
             parent=nodes.get(p);
          }else{
             parent=new TreeNode(p);
             nodes.put(p,parent);
          }

          if(nodes.containsKey(c)){
              child=nodes.get(c);
             }else{
               child=new TreeNode(c);
               nodes.put(c,child);
             }

             if(type==1){
              parent.left=child;
             }else{
              parent.right=child;
             }
          childs.add(c);
    }
    TreeNode root=null;

    for(Integer key:nodes.keySet()){
      if(!childs.contains(key)){
           root=nodes.get(key);
      }
    }
    return root;
}
}